storage/innobase/row/row0merge.cc

```diff
@@ -3549,9 +3555,14 @@ in field data for the tuple
 @param tuple     tuple to be inserted
 @param heap      heap to allocate the memory for the blob storage
 @param blob_file file to handle blob data */
-static dberr_t row_merge_copy_blob_from_file(dtuple_t *tuple, mem_heap_t *heap,
+static dberr_t row_merge_copy_blob_from_file(dtuple_t *tuple,
+                                             dict_index_t *index,
+                                             mem_heap_t *heap,
                                              merge_file_t *blob_file)
 {
+  const uint blob_prefix= dict_table_has_atomic_blobs(index->table)
+                          ? 0
+                          : REC_ANTELOPE_MAX_INDEX_COL_LEN;
   for (ulint i = 0; i < dtuple_get_n_fields(tuple); i++)
```

Comment:
Could the loop start as follows?
```c++
  for (auto i= index->first_user_field(); i < tuple->n_fields; i++)
```
No clustered index field before that may be stored externally.